aten/src/ATen/native/README.md

```diff
@@ -384,6 +390,72 @@ will be automatically differentiated! This can be the case if the function imple
 only calls other operations which are themselves differentiable.  In this
 case, you don't have to write an entry in `tools/autograd/derivatives.yaml`.
 
+### Choosing the right dispatch keyword
+
+After writing a native function in C++, it's important to think about which dispatch keyword
+to use in native_functions.yaml as it gives dispatcher information about backend and autograd support
+of the implementation.
+
+Here're steps to follow to decide the right dispatch keyword:
+
+1. Think about inference, does your kernel work for all backends?
+
+    - No: you're likely providing different kerenls for different backends, e.g. DispatchStub or
+      backend-depenent logics are used in the implementation.
+      Write a dispatch section and enumerate all supported backends and point them to the implementations.
+      ```
+      dispatch:
+        CPU: kernel_cpu
+        CUDA: kernel_cuda
+        QuantizedCPU: kernel_quantized_cpu
+      ```
+
+      [END] Now this op will be called in `CPU/CUDA/QuantizedCPU` backend inference!
```

Comment:
The meaning of `[END]` isn't really defined here; I'd probably suggest just using plain english, e.g., "Stop" or "You're done"